client/components/site-selector-modal/index.jsx

```diff
@@ -35,45 +34,46 @@ class SiteSelectorModal extends Component {
 		// className: class name(s) to be added to the Dialog
 		className: PropTypes.string,
 		// from localize()
-		translate: PropTypes.func.isRequired
+		translate: PropTypes.func.isRequired,
+		// connected props
+		primarySiteId: PropTypes.number
 	}
 
 	constructor( props ) {
 		super( props );
 
-		const primarySite = sitesList.getPrimary();
-		let filteredSites = sitesList.getVisible();
+		const { primarySiteId, visibleSites } = this.props;
+		let filteredSiteIds = visibleSites.map( ( site ) => site.ID );
```

Comment:
This feels selectorish. Since this is done in the constructor, what happens if visibleSites changes?